src/compiler/checker.ts

```diff
@@ -1504,7 +1503,30 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     var lastGetCombinedNodeFlagsResult = NodeFlags.None;
     var lastGetCombinedModifierFlagsNode: Declaration | undefined;
     var lastGetCombinedModifierFlagsResult = ModifierFlags.None;
+    var resolveName = createNameResolver({
+        compilerOptions,
+        requireSymbol,
+        argumentsSymbol,
+        globals,
+        getSymbolOfDeclaration,
+        error,
+        getNodeLinks,
+        lookup: getSymbol,
+        onPropertyWithInvalidInitializer: checkAndReportErrorForInvalidInitializer,
+        onFailedToResolveSymbol,
+        onSuccessfullyResolvedSymbol,
+    });
 
+    const resolveNameForSymbolSuggestion = createNameResolver({

```

Comment:
```suggestion
    var resolveNameForSymbolSuggestion = createNameResolver({
```

Quick nit

Reply:
Ah yes. Reflex to use const. will change.

Reply:
Need like an "only var" rule for these blocks 😄